v2/api/keyvault/customizations/vault_extensions.go

```diff
@@ -184,25 +186,36 @@ func (ex *VaultExtension) handlePurgeThenCreate(
 	return nil
 }
 
+type deletionDetails struct {
+	Exists       bool
+	DeletedVault *armkeyvault.DeletedVault
+}
+
+func deletionDetailsExist(deletedVault armkeyvault.DeletedVault) deletionDetails {
```

Comment:
Maybe use a name that's more declarative at the point of use?

```suggestion
func deletedKeyVaultFound(vault armkeyvault.DeletedVault) deletionDetails {
```

Reply:
Much better.